change(docs): Add publishing crates as part of release checklist (#6909)
ZcashFoundation/zebra: .github/PULL_REQUEST_TEMPLATE/release-checklist.md

- [ ] Publish the pre-release to GitHub using "Publish Release"
- [ ] Delete all the [draft releases from the list of releases](https://github.com/ZcashFoundation/zebra/releases)

## Publish crates

> - [ ] Install `cargo-release` if needed: `cargo install cargo-release`
> - [ ] Check that you are logged into `cargo` as an owner of the crate or a member of the `owners` group
> - [ ] Do a dry-run: `cargo release --workspace -v`
> - [ ] Publish the crates in tree-order: `cargo release --workspace -v --execute`

Reviewer: I'm not sure these steps will work in this order.

#### Version Changes

I think the release steps should go after the `fastmod` steps, and replace most of them. (We can automate the rest later.)

These commands will mark the `tower-batch-control` and `tower-fallback` crates as beta, which is what we want: we don't support them and they don't have a stable API.

Do we also need `cargo login` here? Or is that only checked with `--execute`?

```markdown
<details>

<summary>If you're publishing crates for the first time:</summary>

- [ ] Install `cargo-release` if needed: `cargo install cargo-release`
- [ ] Make sure you are  an owner of the crate or [a member of the `owners` group](https://github.com/orgs/ZcashFoundation/teams/owners) on crates.io

</details>

- [ ] Bump crate versions and do a dry-run:
    - [ ] `cargo release version --verbose --workspace --exclude zebrad beta`
    - [ ] `cargo release version --verbose --package zebrad [ major | minor | patch ]`
    - [ ] `cargo release publish --verbose --workspace --dry-run`
- [ ] Commit the version changes to your release PR branch using `git`: `cargo release commit --verbose --workspace`
```

#### Version Tooling

I think we can delete the `Version Locations` section, and remove the toml changes from `Version Tooling`.

So the remaining steps will look something like:
```
fastmod --fixed-strings '1.0.0' '1.1.0' README.md zebra-network/src/constants.rs book/src/user/docker.md
git add -p
git commit
```

#### Publishing Steps

I think these steps need to go after the release tagging:
```suggestion
- [ ] Run `cargo login`
- [ ] Publish the crates to crates.io: `cargo release publish --verbose --workspace --execute`
```

#### Future Work

We can open tickets to automate the rest of the replacements:
https://github.com/crate-ci/cargo-release/blob/master/docs/faq.md#how-do-i-update-my-readme-or-other-files

And maybe automate the release tag (but this seems more complicated than using GitHub):
https://github.com/crate-ci/cargo-release/blob/master/docs/faq.md#how-do-i-apply-a-setting-only-to-one-crate-in-my-workspace
https://github.com/crate-ci/cargo-release/blob/master/docs/faq.md#how-do-i-customize-my-tagging-in-a-workspace
`git push --tags <name>`

Reviewer: I tried to test these instructions in PR #6908, but I might not have used these exact commands.

Reviewer: I'm not sure what you thought of my other suggestions, here they are as a PR, along with some other cleanups: #6931

Reviewer: Here's the bit I added in that PR:

> I think the release steps should go after the fastmod steps, and replace most of them. (We can automate the rest later.)

Author: #6931 merged into this branch

## Binary Testing

> 

Reviewer: I think we need to test the published crates:
(This can be automated later.)

```suggestion

- [ ] Check that Zebra can be installed from `crates.io`:
      `cargo install --force --version 1.0.0 zebrad && ~/.cargo/bin/zebrad`
```

Reviewer: I tested this command after publishing with PR #6908 and it worked.
